### PR TITLE
fix(gridList): reorder tiles when ngRepeat array changes

### PR DESCRIPTION
**Pull request by tarnas14**
when I tried sorting an array bound to md-grid-list ngRepeat we found that items that didn't change place were displayed incorrectly - this here fixes it (created a demo to prove it - before fix the 4th item would be displayed on 2nd from beginning or end of the md-grid-list after sorting the underlying array)

### CODE REVIEW

**Comment by tarnas14** [comment]
signed the CLA

**Comment by ThomasBurleson** [comment]
@shyndman  - Do you have time to review this PR. I approved let me know and I will merge.
If no time, I will review later. Thx.

**Comment by shyndman** [comment]
Yeah, I'll be able to take a look over the weekend.

File: src/components/gridList/gridList.js [hunk 1]
```diff
@@ -747,8 +760,10 @@ function GridTileDirective($mdMedia) {
     if (angular.isDefined(scope.$parent.$index)) {
       scope.$watch(function() { return scope.$parent.$index; },
         function indexChanged(newIdx, oldIdx) {
-          gridCtrl.removeTile(element, attrs);
-          gridCtrl.addTile(element, attrs, newIdx);
+          if (newIdx === oldIdx) {
+            return;
+          }
+          gridCtrl.reorderTiles();
```
Comment (shyndman) [inline src/components/gridList/gridList.js]: Take the case where someone prepends a tile to the grid -- each of the tiles that already exists in the grid will have this watch trigger, and reorderTiles will be run as many times as there are tiles. This seems like a lot of work just to add a tile.

Comment (shyndman) [inline src/components/gridList/gridList.js]: I'll play around with a couple things. I want to roll in a fix for #2225, since it's related.

Comment (tarnas14) [inline src/components/gridList/gridList.js]: yes, it is computation intensive, I didn't have time to go deeper into the codebase at the time, will probably try to find a better solution today :)

**Comment by shyndman** [comment]
@tarnas14 #2568 will fix your issue.

**Pull request closed without merging**